src/scales/scale.radialLinear.js

```diff
@@ -47,10 +47,300 @@ module.exports = function(Chart) {
+				newRight = pointPosition.x + (textSize.w / 2);
+			} else if (angle < 180) {
+				// text on right side
+				newLeft = pointPosition.x;
+				newRight = pointPosition.x + textSize.w + 5;
+			} else {
+				// text on left side
+				newLeft = pointPosition.x - textSize.w - 5;
+				newRight = pointPosition.x;
+			}
+		}
+
+		function determineYLimits(angle) {
+			if (angle === 90 || angle === 270) {
+				// center Y
+				newTop = pointPosition.y - (textSize.h / 2);
+				newBottom = pointPosition.y + (textSize.h / 2);
+			} else if (angle > 270 || angle < 90) {
+				// text in top half
+				newTop = pointPosition.y - textSize.h - 5;
+				newBottom = pointPosition.y;
+			} else {
+				// text in bottom half
+				newTop = pointPosition.y;
+				newBottom = pointPosition.y + textSize.h + 5;
+			}
+		}
+
+		function updateLimits(angleRadians) {
+			if (newLeft < furthestLeft) {
+				furthestLeft = newLeft;
+				furthestLeftAngle = angleRadians;
+			}
+
+			if (newRight > furthestRight) {
+				furthestRight = newRight;
+				furthestRightAngle = angleRadians;
+			}
+
+			if (newTop < furthestAbove) {
+				furthestAbove = newTop;
+				furthestAboveAngle = angleRadians;
+			}
+
+			if (newBottom > furthestBelow) {
+				furthestBelow = newBottom;
+				furthestBelowAngle = angleRadians;
+			}
+		}
+
+		var valueCount = getValueCount(scale);
+		for (i = 0; i < valueCount; i++) {
+			pointPosition = scale.getPointPosition(i, largestPossibleRadius);
+			textSize = measureLabelSize(scale.ctx, plFont.size, scale.pointLabels[i] || '');
+			scale._pointLabelSizes[i] = textSize;
+
+			// Add quarter circle to make degree 0 mean top of circle
+			var angleRadians = scale.getIndexAngle(i);
+			var angle = helpers.toDegrees(angleRadians) % 360;
+			determineXLimits(angle);
+			determineYLimits(angle);
+			updateLimits(angleRadians);
+		}
+
+		function determineReductions() {
+			var radiusReductionLeft = furthestLeft / Math.sin(furthestLeftAngle);
+			var radiusReductionRight = Math.max(furthestRight - scale.width, 0) / Math.sin(furthestRightAngle);
+			var radiusReductionTop = -furthestAbove / Math.cos(furthestAboveAngle);
+			var radiusReductionBottom = -Math.max(furthestBelow - scale.height, 0) / Math.cos(furthestBelowAngle);
+
+			radiusReductionLeft = helpers.isNumber(radiusReductionLeft) ? radiusReductionLeft : 0;
+			radiusReductionRight = helpers.isNumber(radiusReductionRight) ? radiusReductionRight : 0;
+			radiusReductionTop = helpers.isNumber(radiusReductionTop) ? radiusReductionTop : 0;
+			radiusReductionBottom = helpers.isNumber(radiusReductionBottom) ? radiusReductionBottom : 0;
+
+			scale.drawingArea = Math.min(
+				Math.round(largestPossibleRadius - (radiusReductionLeft + radiusReductionRight) / 2),
+				Math.round(largestPossibleRadius - (radiusReductionTop + radiusReductionBottom) / 2));
+			scale.setCenterPoint(radiusReductionLeft, radiusReductionRight, radiusReductionTop, radiusReductionBottom);
+		}
+		determineReductions();
```

Comment:
That was to make eslint happy otherwise the function is too long and too complex